packages/flutter/lib/src/material/navigation_drawer.dart

```diff
@@ -0,0 +1,416 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'color_scheme.dart';
+import 'colors.dart';
+import 'drawer.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'material_state.dart';
+import 'navigation_bar.dart';
+import 'navigation_drawer_theme.dart';
+import 'text_theme.dart';
+import 'theme.dart';
+
+/// Material 3 Navigation Drawer component.
+///
+/// Navigation drawers offer a persistent and convenient way to switch between
+/// primary destinations in an app.
+///
+/// The style for the icons and text are not affected by parent
+/// [DefaultTextStyle]s or [IconTheme]s but rather controlled by parameters or
+/// the [NavigationDrawerThemeData].
+///
+/// This widget holds a collection of destinations ([NavigationDrawerDestination]s) and
+/// customized widgets, e.g. customized headlines, section titles, dividers.
+///
+/// Only [NavigationDrawerDestination] is counted in index.
+///
+/// {@tool dartpad}
+/// This example shows a [NavigationDrawer] as it is used within a [Scaffold]
+/// widget. The [NavigationDrawer] has headline widget, divider widget and three
+/// [NavigationDrawerDestination] widgets. The initial [selectedIndex] is set to index
+/// 0. The [onDestinationSelected] callback changes the selected item's index and displays
+/// a corresponding widget in the body of the [Scaffold].
+///
+/// ** See code in examples/api/lib/material/navigation_drawer/navigation_drawer.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [Scaffold.drawer], where one specifies a [Drawer] so that it can be
+///    shown.
+///  * [Scaffold.of], to obtain the current [ScaffoldState], which manages the
+///    display and animation of the drawer.
+///  * [ScaffoldState.openDrawer], which displays its [Drawer], if any.
+///  * <https://material.io/design/components/navigation-drawer.html>
+class NavigationDrawer extends StatelessWidget {
+  /// Creates a Material 3 Navigation Drawer component.
+  const NavigationDrawer({
+    super.key,
+    required this.children,
+    this.backgroundColor,
+    this.surfaceTintColor,
+    this.elevation,
+    this.onDestinationSelected,
+    this.selectedIndex = 0,
+  });
+
+  /// Sets the color of the [Material] that holds all of the [NavigationDrawer]'s
+  /// contents.
+  ///
+  /// If this is null, then [NavigationDrawerThemeData.backgroundColor] is used.
+  /// If that is also null, then it falls back to [Material]'s default.
+  final Color? backgroundColor;
+
+  /// Sets the color of the [Material] that holds all of the [NavigationDrawer]'s
+  /// contents.
+  ///
+  /// If this is null, then [NavigationDrawerThemeData.surfaceTintColor] is used.
+  /// If that is also null, then it falls back to [Material]'s default.
+  final Color? surfaceTintColor;
```

Comment:
We should probably also add a `shadowColor` for this as well even if we normally set it to null or transparent for the M3 case. Whenever we expose the `elevation` we should also include the `surfaceTintColor` and `shadowColor`.

Reply:
Done